ProcessLib/ThermoMechanics/ThermoMechanicsFEM.h

```diff
@@ -276,13 +284,13 @@ class ThermoMechanicsLocalAssembler
                     .noalias() = N;
 
             // calculate real density
-            // rho_s * (V_0 + dV) = rho_sr * V_0
+            // rho_s_{n+1} * (V_{n} + dV) = rho_s_n * V_n
             // dV = 3 * alpha * dT * V_0
-            // rho_s = rho_sr / (1 + 3 * alpha * dT )
+            // rho_s_{n+1} = rho_s_n / (1 + 3 * alpha * dT )
             // see reference solid density description for details.
-            auto const rho_sr =
-                _process_data.reference_solid_density(t, x_position)[0];
-            double const rho_s = rho_sr / (1 + 3 * linear_thermal_strain);
+            double const rho_s = _ip_data[ip].solid_density_prev /
+                                 (1 + 3 * linear_thermal_strain);
```

Comment:
Not introduced in this PR, but is the number 3 related to the space dimension?

Reply:
We set input of the thermal expansion, alpha_T,  as line thermal expansion. The means it is the  thermal expansion for strain component as 
d epsilon^T_{xx} = alpha_T dT
d epsilon^T_{yy} = alpha_T dT
d epsilon^T_{zz} = alpha_T dT

For the volume change due to the thermal expansion, we have
d epsilon^T_{xx} + d epsilon^T_{yy} + d epsilon^T_{zz} = 3 alpha_T dT.
This is valid for 1D, 2D (either plane strain or plane stress) and 3D problem.

Reply:
OK, thanks. But in shouldn't it be different in plane strain, when all deformation is confined to two dimensions, or am I confusing something, here?

Reply:
In 2D plane strain, ![e](https://user-images.githubusercontent.com/1343839/43575868-2f1ac6b8-9648-11e8-8fa0-28b02b0d70d0.gif)
which means
![e2](https://user-images.githubusercontent.com/1343839/43575892-414180a2-9648-11e8-8918-a660100d0dc1.gif)
In 2D plane stress (we do not need it)
\dot \epsilon_{zz} != 0, we still have
![e3](https://user-images.githubusercontent.com/1343839/43575927-54f2dcb8-9648-11e8-8ab3-240fc7fb68bd.gif)
where the superscript e means elastic, and T mean thermal.

Reply:
Your confusing leads a correction to volume thermal strain under  2D plane strain condition. Thanks. I will  change it to `1 + DisplacementDim  * linear_thermal_strain`.

Reply:
I withdraw my last answer.

Reply:
As written two lines above the "offending" one I probably should `// see reference solid density description for details.`
Thanks for your explanations so far. If I'll have some time, I'll dive into the details to fully understand the formula.